apps/src/templates/sectionsRefresh/CoteacherSettings.jsx

```diff
@@ -0,0 +1,133 @@
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import i18n from '@cdo/locale';
+
+import styles from './coteacher-settings.module.scss';
+import Button from '../Button';
+import {Figcaption} from '@cdo/apps/componentLibrary/typography';
+import FontAwesome from '../FontAwesome';
+import classNames from 'classnames';
+import {isEmail} from '@cdo/apps/util/formatValidation';
+
+export default function CoteacherSettings({
+  sectionInstructors,
+  addCoteacher,
+  coteachersToAdd,
+}) {
+  const [inputValue, setInputValue] = useState('');
+  const [addError, setAddError] = useState('');
+
+  const handleInputChange = event => {
+    setInputValue(event.target.value);
+  };
+
+  const handleAddEmail = e => {
+    e.preventDefault();
+    const newEmail = inputValue;
+    if (newEmail === '') {
+      setAddError(i18n.coteacherAddNoEmail());
+      return;
+    }
+    if (!isEmail(newEmail)) {
+      setAddError(i18n.coteacherAddInvalidEmail({email: newEmail}));
+      return;
+    }
+    if (
+      (sectionInstructors &&
+        sectionInstructors.some(
+          instructor => instructor.instructorEmail === newEmail
+        )) ||
+      coteachersToAdd.some(email => email === newEmail)
+    ) {
+      setAddError(i18n.coteacherAddAlreadyExists({email: newEmail}));
+      return;
+    }
+
+    addCoteacher(newEmail);
+    setAddError('');
+    setInputValue('');
+  };
+
+  const handleSubmitAddEmail = e => {
+    if (e.key === 'Enter') {
+      e.preventDefault();
+      handleAddEmail(e);
+    }
+  };
+
+  // coteacher count is teachers to add + the existing teachers - the primary teacher.
+  // If sectionInstructors is empty, we still want to count the primary teacher.
+  const sectionInstructorsCount = sectionInstructors
+    ? sectionInstructors.length
+    : 1;
+  const numCoteachers = sectionInstructorsCount + coteachersToAdd.length - 1;
```

Comment:
Seems a bit clearer.

```suggestion
  // coteacher count is teachers to add + the existing teachers - the primary teacher.
  // If sectionInstructors is populated, we assume one of them is the primary teacher.
  const existingCoteachers = sectionInstructors
    ? sectionInstructors.length - 1
    : 0;
  const numCoteachers = existingCoteachers + coteachersToAdd.length - 1;
```